micrometer-core/src/main/java/io/micrometer/core/instrument/distribution/AbstractTimeWindowHistogram.java

```diff
@@ -166,25 +171,6 @@ private ValueAtPercentile[] takeValueSnapshot() {
         return values;
     }
 
-    private CountAtBucket[] takeCountSnapshot() {
-        if (!distributionStatisticConfig.isPublishingHistogram()) {
-            return null;
-        }
-
-        final Set<Double> monitoredValues = distributionStatisticConfig
-            .getHistogramBuckets(supportsAggregablePercentiles);
```

Comment:
This logic of getting the buckets again to include in the snapshot was problematic for the goal of always having an infinity bucket (in the case of Stackdriver), since this relies only on the logic in `DistributionStatisticConfig` and we do not know at this level (`AbstractTimeWindowHistogram`) if we want to include the infinity bucket or not. Hence this was removed to avoid trying to keep the buckets in the concrete histogram type (e.g. `TimeWindowFixedBoundaryHistogram`) to match the buckets returned here when `Histogram#takeSnapshot` is called. With the new package-private method `countsAtBuckets` we let the concrete implementation determine what buckets to return rather than from here passing the buckets. We could even get rid of the `countsAtValues` method, I think.

Reply:
Indeed, `Iterator<CountAtBucket> countsAtValues(Iterator<Double> values)` might be unnecessary while we are also having `CountAtBucket[] countsAtBuckets()`. I'm not sure though if there is a use-case of `countsAtValues`, where not all the bucket boundaries are passed.

Reply:
It's package-private, so we can check all the usage. AFAIR, it's always `distributionStatisticConfig.getHistogramBuckets` being called, until this PR deviated to add the infinity bucket. We previously had a regression where the buckets didn't match - see https://github.com/micrometer-metrics/micrometer/pull/3922.

Reply:
I've removed it now as I refactored things in `TimeWindowPercentileHistogram`